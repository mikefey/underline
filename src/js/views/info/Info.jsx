import React from 'react';
import PropTypes from 'prop-types';
import Header from 'ui/header/Header.jsx';
import CloseButton from 'ui/close-button/CloseButton.jsx';
import style from './Info.scss';

/**
 * Info component
 * @param {Object} props - The component props
 */
const Info = (props) => {
  const wrapperClass = props.infoState.show
    ? style.wrapperShowing
    : style.wrapper;

  return (
    <div className={wrapperClass}>
      <Header
        rightButton={
          <CloseButton
            clickHandler={props.hideInfo}
          />
        }
        text='Underline'
      />
      <div className={style.contentWrapper}>
        <div className={style.infoBlock}>
          <div className={style.title}>Contact</div>
          <div className={style.content}>Mike Fey (<a href='mailto:mikefey@protonmail.com'>mikefey@protonmail.com</a>)</div>
        </div>
        <div className={style.infoBlock}>
          <div className={style.title}>Bugs</div>
          <div className={style.content}>Please open an issue or pull request at <a href='https://github.com/mikefey/underline'>https://github.com/mikefey/underline</a></div>
        </div>
        <div className={style.infoBlock}>
          <div className={style.title}>License</div>
          <div className={style.content}>This software is covered under the <a href='https://www.gnu.org/licenses/gpl-3.0.en.html'>GNU General Public License v3.0</a></div>
        </div>
        <div className={style.infoBlock}>
          <div className={style.title}>"Add" button icon credit:</div>
          <div className={style.content}>Icon made by <a href='http://picol.org/'>Picol</a> from <a href='http://flaticon.com'>flaticon.com</a></div>
        </div>
        <div className={style.infoBlock}>
          <div className={style.title}>Privacy</div>
          <div className={style.content}>
            <p className='statement'>Your privacy is critically important. Your personal information is not stored on our servers.</p>
            <p>It is Underline's policy to respect your privacy regarding any information we may collect while operating our app.</p>
            <strong>Website/App Visitors</strong>
            <p>Like most website/app operators, Underline collects non-personally-identifying information of the sort that web browsers and servers typically make available, such as the browser type, language preference, referring site, and the date and time of each visitor request. Underline's purpose in collecting non-personally identifying information is to better understand how Underline's visitors use the app. From time to time, Underline may release non-personally-identifying information in the aggregate, e.g., by publishing a report on trends in the usage of the app.</p>
            <strong>Gathering of Personally-Identifying Information</strong>
            <p>Certain visitors to Underline website/app choose to interact with Underline in ways that require Underline to gather personally-identifying information. The amount and type of information that Underline gathers depends on the nature of the interaction. Underline collects such information only insofar as is necessary or appropriate to fulfill the purpose of the visitor's interaction with Underline. Underline does not disclose personally-identifying information other than as described below. And visitors can always refuse to supply personally-identifying information, with the caveat that it may prevent them from engaging in certain website/app-related activities.</p>
            <strong>Aggregated Statistics</strong>
            <p>Underline may collect statistics about the behavior of visitors to its websites and app. However, Underline does not disclose personally-identifying information other than as described below.</p>
            <strong>IndexedDB</strong>
            <p>IndexedDB is a database that websites and apps use to store information on a visitor's computer. The visitor's browser provides that data to the website each time the visitor returns. Underline uses IndexedDB to store the app's data on the vistor's computer  .</p>
            <strong>Privacy Policy Changes</strong>
            <p>Although most changes are likely to be minor, Underline may change its Privacy Policy from time to time, and in Underline's sole discretion. Underline encourages visitors to frequently check this page for any changes to its Privacy Policy.</p>
            <p>Modified from <a href='http://automattic.com/'>Automattic</a> Privacy Policy</p>
          </div>
        </div>
      </div>
    </div>
  );
};


/**
 * PropTypes
 * @prop {Function} hideInfo - Function to dispatch a HIDE_INFO action
 * @prop {Object} infoState - The info state in the store
 */
Info.propTypes = {
  hideInfo: PropTypes.func.isRequired,
  infoState: PropTypes.object.isRequired,
};

export default Info;
