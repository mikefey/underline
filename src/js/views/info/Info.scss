@import '~colors.scss';
@import '~variables.scss';

.wrapper {
  background-color: $white;
  display: none;
  float: left;
  height: 100%;
  position: fixed;
  width: 100%;
  z-index: 750;
}

.wrapperShowing {
  composes: wrapper;
  display: block;
}

.contentWrapper {
  float: left;
  height: 100%;
  max-width: $max-content-width;
  overflow-y: scroll;
  -webkit-overflow-scrolling: touch;
  padding: 80px 20px;
  position: relative;
  text-align: left;
  width: 100%;
}

.infoBlock {
  float: left;
  line-height: 1.5em;
  margin-bottom: 20px;
  width: 100%;
}

.title,
strong {
  float: left;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 24px;
  font-weight: bold;
  line-height: 1.5em;
  width: 100%;
}

strong {
  font-size: 18px;
}

.content {
  font-size: 20px;
  line-height: 1.5em;

  a {
    color: $rose;
  }
}
