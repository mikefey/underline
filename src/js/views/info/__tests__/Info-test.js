import test from 'ava';
import React from 'react';
import { renderToString } from 'react-dom/server';

import Info from './../Info.jsx';

test('Info component', (t) => {
  const infoState = {
    show: true,
  };
  const hideInfo = () => {};

  const tree = renderToString(<Info
    infoState={infoState}
    hideInfo={hideInfo}
  />);

  t.snapshot(tree);
});
