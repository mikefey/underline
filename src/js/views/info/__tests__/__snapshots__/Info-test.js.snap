// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component 1`] = `"<div class=\\"Info__wrapperShowing___oSzkgcvJGgm_tAV Info__wrapper___BoibkEk8lqvqZeV\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-1920563339\\"><div class=\\"Header__wrapper___2PTLf_nkW7S2HEU\\" data-reactid=\\"2\\"><div class=\\"Header__content___2VIdu565Foup5Pb\\" data-reactid=\\"3\\"><h2 class=\\"Header__title___1_ebCiqkttbS44m\\" data-reactid=\\"4\\">Underline</h2><button class=\\"CloseButton__wrapper___2i9TByxUopDp0cZ\\" data-reactid=\\"5\\"><div class=\\"CloseButton__icon___3Schb-DMASl5XVC\\" data-reactid=\\"6\\"><div class=\\"CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\" data-reactid=\\"7\\"></div><div class=\\"CloseButton__iconLineTwo___3dpb1YfW7QXAzBv CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\" data-reactid=\\"8\\"></div></div></button></div></div><div class=\\"Info__contentWrapper___23y7LKZxWPqLQh4\\" data-reactid=\\"9\\"><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"10\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"11\\">Contact</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"12\\"><!-- react-text: 13 -->Mike Fey (<!-- /react-text --><a href=\\"mailto:mikefey@protonmail.com\\" data-reactid=\\"14\\">mikefey@protonmail.com</a><!-- react-text: 15 -->)<!-- /react-text --></div></div><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"16\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"17\\">Bugs</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"18\\"><!-- react-text: 19 -->Please open an issue or pull request at <!-- /react-text --><a href=\\"https://github.com/mikefey/underline\\" data-reactid=\\"20\\">https://github.com/mikefey/underline</a></div></div><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"21\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"22\\">License</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"23\\"><!-- react-text: 24 -->This software is covered under the <!-- /react-text --><a href=\\"https://www.gnu.org/licenses/gpl-3.0.en.html\\" data-reactid=\\"25\\">GNU General Public License v3.0</a></div></div><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"26\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"27\\">&quot;Add&quot; button icon credit:</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"28\\"><!-- react-text: 29 -->Icon made by <!-- /react-text --><a href=\\"http://picol.org/\\" data-reactid=\\"30\\">Picol</a><!-- react-text: 31 --> from <!-- /react-text --><a href=\\"http://flaticon.com\\" data-reactid=\\"32\\">flaticon.com</a></div></div><div class=\\"Info__infoBlock___1dpxEUTI23jwgeF\\" data-reactid=\\"33\\"><div class=\\"Info__title___PrfjZjNeHKd2QcF\\" data-reactid=\\"34\\">Privacy</div><div class=\\"Info__content___1dKJlGjnm2X2-Gb\\" data-reactid=\\"35\\"><p class=\\"statement\\" data-reactid=\\"36\\">Your privacy is critically important. Your personal information is not stored on our servers.</p><p data-reactid=\\"37\\">It is Underline&#x27;s policy to respect your privacy regarding any information we may collect while operating our app.</p><strong data-reactid=\\"38\\">Website/App Visitors</strong><p data-reactid=\\"39\\">Like most website/app operators, Underline collects non-personally-identifying information of the sort that web browsers and servers typically make available, such as the browser type, language preference, referring site, and the date and time of each visitor request. Underline&#x27;s purpose in collecting non-personally identifying information is to better understand how Underline&#x27;s visitors use the app. From time to time, Underline may release non-personally-identifying information in the aggregate, e.g., by publishing a report on trends in the usage of the app.</p><strong data-reactid=\\"40\\">Gathering of Personally-Identifying Information</strong><p data-reactid=\\"41\\">Certain visitors to Underline website/app choose to interact with Underline in ways that require Underline to gather personally-identifying information. The amount and type of information that Underline gathers depends on the nature of the interaction. Underline collects such information only insofar as is necessary or appropriate to fulfill the purpose of the visitor&#x27;s interaction with Underline. Underline does not disclose personally-identifying information other than as described below. And visitors can always refuse to supply personally-identifying information, with the caveat that it may prevent them from engaging in certain website/app-related activities.</p><strong data-reactid=\\"42\\">Aggregated Statistics</strong><p data-reactid=\\"43\\">Underline may collect statistics about the behavior of visitors to its websites and app. However, Underline does not disclose personally-identifying information other than as described below.</p><strong data-reactid=\\"44\\">IndexedDB</strong><p data-reactid=\\"45\\">IndexedDB is a database that websites and apps use to store information on a visitor&#x27;s computer. The visitor&#x27;s browser provides that data to the website each time the visitor returns. Underline uses IndexedDB to store the app&#x27;s data on the vistor&#x27;s computer  .</p><strong data-reactid=\\"46\\">Privacy Policy Changes</strong><p data-reactid=\\"47\\">Although most changes are likely to be minor, Underline may change its Privacy Policy from time to time, and in Underline&#x27;s sole discretion. Underline encourages visitors to frequently check this page for any changes to its Privacy Policy.</p><p data-reactid=\\"48\\"><!-- react-text: 49 -->Modified from <!-- /react-text --><a href=\\"http://automattic.com/\\" data-reactid=\\"50\\">Automattic</a><!-- react-text: 51 --> Privacy Policy<!-- /react-text --></p></div></div></div></div>"`;
